### PR TITLE
Add blog post for cancellation in JS

### PR DESCRIPTION
**Pull request by bterlson**

### CODE REVIEW

**Comment by bterlson** [comment]
@jongio in the mangeld #1226 pointed out two things I want to make sure are reflected in the discussion

1. The point about separation of concerns between controller and signal isn't clear. Any suggestions on how to improve would be appreciated!
2. In the intro, we could use a bulleted list instead of prose (and maybe add more scenarios besides). Anyone feel strongly here?

File: _posts/2020-04-21-js-cancellation.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+The new Azure SDK libraries for JavaScript and TypeScript have adopted abort signals for just these purposes.
+
+## AbortController &amp; AbortSignal
+
+`AbortController` and `AbortSignal` are [standard features in the browser](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) and are used with the [`fetch` API](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) to abort in-progress network requests. The controller is responsible for triggering the cancellation, and signals are responsible for notifying when a cancellation has been triggered. This separation of concerns enables you to safely pass an abort signal without delegating the ability to cancel the signal as well.
```
Comment (jongio) [inline _posts/2020-04-21-js-cancellation.md:16]: Consider:

This separation of concerns enables you to safely pass the abort signal to a other classes and not be concerned with it unintentionally cancelling the request.

Comment (bterlson) [reply _posts/2020-04-21-js-cancellation.md:16]: Do you think the point is even worth keeping? I feel like it doesn't add much.

Comment (bterlson) [reply _posts/2020-04-21-js-cancellation.md:16]: I like that. I'll replace classes with "other parts of your application", though. Not everyone likes classes 😀

**Comment by bterlson** [comment]
@adrianhall can you check out the new intro paragraph?

File: _posts/2020-04-21-js-cancellation.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
+---
+title: How to use abort signals to cancel operations in the Azure SDK for JavaScript/TypeScript
+layout: post
+date: 21 apr 2020
+sidebar: releases_sidebar
+author_github: bterlson
+repository: azure/azure-sdk
+---
+
+Cancelling in-progress network operations is critical for applications to maintain responsiveness and avoid doing work that isn't needed anymore. There are many situations where you want to cancel on-going work, for example when you are:
```
Comment (adrianhall) [inline _posts/2020-04-21-js-cancellation.md:10]: There are many situations where you want to cancel on-going work.  For example, when you are:

File: _posts/2020-04-21-js-cancellation.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+* downloading a large file from Azure Storage and the user wants to cancel the download
+* processing events from an Event Hub and you need to gracefully shut down in order to scale down or to apply updates
+* running an operation that might take a while and you don't want to wait longer than a fixed amount of time
+* running multiple operations in parallel and getting one result makes the other results irrelevant
+
+The new Azure SDK libraries for JavaScript and TypeScript have adopted abort signals situations like these.
```
Comment (adrianhall) [inline _posts/2020-04-21-js-cancellation.md:17]: ... adopted abort signals for situations like these.  Also, remove "new"

File: _posts/2020-04-21-js-cancellation.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+## AbortController &amp; AbortSignal
+
+`AbortController` and `AbortSignal` are [standard features in the browser](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) and are used with the [`fetch` API](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) to abort in-progress network requests. The controller is responsible for triggering the cancellation, and signals are responsible for notifying when a cancellation has been triggered. This separation of concerns enables you to safely pass the abort signal to other parts of your application without worrying about it unintentionally cancelling the request.
+
+If you're only targeting fairly modern browsers, you can use the built-in `AbortController` and `AbortSignal` and everything will work fine. If you're targeting Node.js, or if you want to take advantage of linked signals or other features that I'll cover later in this post, you can use the implementation in the Azure SDK found in the [`@azure/abort-controller` npm package](https://www.npmjs.com/package/@azure/abort-controller).
```
Comment (adrianhall) [inline _posts/2020-04-21-js-cancellation.md:23]: Potentially link to `https://caniuse.com/#feat=abortcontroller`?

File: _posts/2020-04-21-js-cancellation.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+To abort an in-progress request such as an upload or download in storage, create a new `AbortController` and pass its signal into the method you might want to cancel later:
```
Comment (adrianhall) [inline _posts/2020-04-21-js-cancellation.md:25]: I had to read this twice - suggest "To abort an in-progress request (such as a download from Azure Storage), ...

**Pull request merged**